src/Microsoft.VisualStudio.Threading/JoinableTaskDependencyGraph.cs

```diff
@@ -502,7 +502,11 @@ internal static void OnSynchronousTaskEndToBlockWaiting(JoinableTask syncTask)
                     lock (syncTask.Factory.Context.SyncContextLock)
                     {
                         // Remove itself from the tracking list, after the task is completed.
-                        RemoveDependingSynchronousTask(syncTask, syncTask, force: true);
+                        var syncTaskItem = (IJoinableTaskDependent)syncTask;
```

Comment:
Your `(IJoinableTaskDependent)` cast makes me wonder if it may ever fail with `InvalidCastException`. But if `JoinableTask` directly implements this interface, you can avoid the cast and make it clear in a review that this can never fail (as well as using the compiler to draw attention to this line if that assumption is ever false) by rewriting this line as:

```cs
IJoinableTaskDependent syncTaskItem = syncTask;
```

Reply:
makes sense.